src/inet_address.rs

```diff
@@ -38,3 +42,63 @@ impl InetAddress {
         }
     }
 }
+
+pub trait InetAddressExtManual {
+    fn to_bytes<'a>(&'a self) -> Option<InetAddressBytes<'a>>;
+}
+
+impl<O: IsA<InetAddress>> InetAddressExtManual for O {
+    /// Returns `None` in case the address has a native size different than 4 and 16.
+    fn to_bytes<'a>(&'a self) -> Option<InetAddressBytes<'a>> {
+        let size = self.get_native_size();
+        unsafe {
+            let bytes = gio_sys::g_inet_address_to_bytes(self.as_ref().to_glib_none().0);
+            if size == 4 {
+                // The `try_into` is used to convert from `&[u8]` to `&[u8; 4]`.
+                Some(InetAddressBytes::V4(
+                    from_raw_parts(bytes, 4)
+                        .try_into()
+                        .expect("failed to convert to array"),
```

Comment:
```suggestion
                Some(InetAddressBytes::V4(&*(bytes as *const [u8; 4]),
```

Should work equally well and less line noise. If it doesn't, how does it fail?

Reply:
Aren't you creating a temporary reference by doing so? We're moving a variable here, not just using a function, so I have a doubt...

Reply:
In what sense? `*const [u8; 4]` is equivalent to `&[u8;4]` and either of those are equivalent to `*const u8` just that there is static type system information about the length.

Reply:
That's how the `try_into()` you're using here works internally too, just that we can skip all these redundant steps (all `try_into()` is checking if the size is 4, which you made sure by passing 4 to `from_raw_parts()`)

Reply:
I'm definitely not up-to-date about references/pointers cast...

Reply:
https://doc.rust-lang.org/src/core/array/mod.rs.html#157-163